internal/stacks/stackconfig/provider_requirements.go

```diff
@@ -201,6 +202,13 @@ func decodeProviderRequirementsBlock(block *hcl.Block) (*ProviderRequirements, t
 
 func (pr *ProviderRequirements) ProviderForLocalName(localName string) (addrs.Provider, bool) {
 	if pr == nil {
+		// Check whether the provider *might be* built-in.
+		// There might be special occasions where localName corresponds to a
+		// built-in provider names. In that case, customers will see an error
+		// occurring later in the process.
+		if _, ok := builtinProviders.BuiltInProviders()[localName]; ok {
+			return addrs.Provider{}, true
+		}
 		return addrs.Provider{}, false
 	}
 	obj, ok := pr.Requirements[localName]
```

Comment:
I'm surprised you didn't need to change here as well - what if the provider requirements block was written but it doesn't have the builtin provider present? Perhaps adding another test where there is a different provider defined in the required providers but not the builtin, but the builtin is then referred to would expose this?

There could also be another explanation as to why this isn't required!

Reply:
You're surprised for valid reasons. I made an oversight. This should be fixed here 45136b4bed92d0395d6399a8862a75c30d555f9f

Thanks for catching on this!